How to get the cluster_provider to work

Hi Alessio Dionisi,

Great that you made a K0s provider in TF. I use TF a lot and work with K0s daily, but I'm lacking go skills.. ;-(

Can you document and/or tell me how to get the k0s_cluster provider working (including how to deal with go).
I have go installed but it rains errors as per my lack of knowledge....

[ec2-user@ip-172-31-24-202 cluster_resource]$ vi cluster_resource.go
[ec2-user@ip-172-31-24-202 cluster_resource]$ go build
cluster_resource.go:7:2: cannot find package "github.com/hashicorp/terraform-plugin-framework/diag" in any of:
	/usr/lib/golang/src/github.com/hashicorp/terraform-plugin-framework/diag (from $GOROOT)
	/home/ec2-user/go/src/github.com/hashicorp/terraform-plugin-framework/diag (from $GOPATH)
cluster_resource.go:8:2: cannot find package "github.com/hashicorp/terraform-plugin-framework/tfsdk" in any of:
	/usr/lib/golang/src/github.com/hashicorp/terraform-plugin-framework/tfsdk (from $GOROOT)
	/home/ec2-user/go/src/github.com/hashicorp/terraform-plugin-framework/tfsdk (from $GOPATH)
cluster_resource.go:9:2: cannot find package "github.com/hashicorp/terraform-plugin-framework/types" in any of:
	/usr/lib/golang/src/github.com/hashicorp/terraform-plugin-framework/types (from $GOROOT)
	/home/ec2-user/go/src/github.com/hashicorp/terraform-plugin-framework/types (from $GOPATH)
cluster_resource.go:10:2: cannot find package "github.com/hashicorp/terraform-plugin-go/tftypes" in any of:
	/usr/lib/golang/src/github.com/hashicorp/terraform-plugin-go/tftypes (from $GOROOT)
	/home/ec2-user/go/src/github.com/hashicorp/terraform-plugin-go/tftypes (from $GOPATH)
cluster_resource.go:11:2: cannot find package "github.com/k0sproject/dig" in any of:
	/usr/lib/golang/src/github.com/k0sproject/dig (from $GOROOT)
	/home/ec2-user/go/src/github.com/k0sproject/dig (from $GOPATH)
cluster_resource.go:12:2: cannot find package "github.com/k0sproject/k0sctl/phase" in any of:
	/usr/lib/golang/src/github.com/k0sproject/k0sctl/phase (from $GOROOT)
	/home/ec2-user/go/src/github.com/k0sproject/k0sctl/phase (from $GOPATH)
cluster_resource.go:13:2: cannot find package "github.com/k0sproject/k0sctl/pkg/apis/k0sctl.k0sproject.io/v1beta1" in any of:
	/usr/lib/golang/src/github.com/k0sproject/k0sctl/pkg/apis/k0sctl.k0sproject.io/v1beta1 (from $GOROOT)
	/home/ec2-user/go/src/github.com/k0sproject/k0sctl/pkg/apis/k0sctl.k0sproject.io/v1beta1 (from $GOPATH)
cluster_resource.go:14:2: cannot find package "github.com/k0sproject/k0sctl/pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster" in any of:
	/usr/lib/golang/src/github.com/k0sproject/k0sctl/pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster (from $GOROOT)
	/home/ec2-user/go/src/github.com/k0sproject/k0sctl/pkg/apis/k0sctl.k0sproject.io/v1beta1/cluster (from $GOPATH)
cluster_resource.go:15:2: cannot find package "github.com/k0sproject/rig" in any of:
	/usr/lib/golang/src/github.com/k0sproject/rig (from $GOROOT)
	/home/ec2-user/go/src/github.com/k0sproject/rig (from $GOPATH)
cluster_resource.go:16:2: cannot find package "gopkg.in/yaml.v2" in any of:
	/usr/lib/golang/src/gopkg.in/yaml.v2 (from $GOROOT)
	/home/ec2-user/go/src/gopkg.in/yaml.v2 (from $GOPATH)

--- a/internal/provider/cluster_resource.go
+++ b/internal/provider/cluster_resource.go
@@ -359,6 +359,7 @@ func getK0sctlConfig(data *ClusterResourceModel) *k0sctl_v1beta1.Cluster {
 					Address: host.SSH.Address.ValueString(),
 					Port:    int(host.SSH.Port.ValueInt64()),
 					User:    host.SSH.User.ValueString(),
+					KeyPath: host.SSH.KeyPath.ValueString(),
 				},
 			},
 			Role:             host.Role.ValueString(),

--- a/internal/provider/cluster_resource.go
+++ b/internal/provider/cluster_resource.go
@@ -359,7 +359,7 @@ func getK0sctlConfig(data *ClusterResourceModel) *k0sctl_v1beta1.Cluster {
 					Address: host.SSH.Address.ValueString(),
 					Port:    int(host.SSH.Port.ValueInt64()),
 					User:    host.SSH.User.ValueString(),
-					KeyPath: host.SSH.KeyPath.ValueString(),
+					KeyPath: host.SSH.KeyPath.ValueStringPointer(),
 				},
 			},
 			Role:             host.Role.ValueString(),

